front-end-web/src/Components/Pages/ProfilePage.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  Button,
+  Typography,
+  TextField,
+  Dialog,
+  DialogContentText,
+  DialogContent,
+  DialogActions,
+  DialogTitle
+} from "@material-ui/core";
+import React, { Fragment } from "react";
+import firebase from "firebase";
+import firebaseApp from "../../scripts/firebaseConfig";
+import { NotificationMessage } from "../../@Types";
+declare interface ProfileProps {
+  openPage: boolean;
+  handleToggleProfile: () => void;
+
+  handleNewNotification: (NewNotification: NotificationMessage) => void;
+}
+
+/**
+ * TODO: Add Documentation
+ */
+
+let newEmail: string = "";
+let newPassword: string = "";
+let newPasswordCheck: string = "";
+let newName: string = "";
+let currentName: string = "";
+let currentEmail: string = "";
+
+firebaseApp.auth();
+
+const ProfilePage: React.FunctionComponent<ProfileProps> = ({
+  openPage,
+  handleToggleProfile,
+  handleNewNotification
```

Comment:
I would add `currentUser` here. The current user is already stored in the React state, so it is faster to get from the state than it is from Firebase each time.